File: anarc08b.py
import sys
import psyco
psyco.full()
def pri():
	try: s=sys.stdin.readline()
	except: return
	s = s.replace('\n','')
	while(s!="BYE"):
		i=0
        	num1=0
		num2=0
		num=0;
		while(s[i] is not '+' and i<len(s)):
			c=""
			for j in range(0,3):
				if i+j < len(s):
					c=c+s[i+j];
				else:
					break
	#	print c
			if(c=="063"):
				num=num*10;
			elif(c=="010"):
				num=num*10+1;
			elif(c=="093"):
				num=num*10+2;
			elif(c=="079"):
				num=num*10+3;
			elif(c=="106"):
				num=num*10+4;
			elif(c=="103"):
				num=num*10+5;
			elif(c=="119"):
				num=num*10+6;
			elif(c=="011"):
				num=num*10+7;
			elif(c=="127"):
				num=num*10+8;
			elif(c=="111"):
				num=num*10+9;			
			i=i+3;
		i= i + 1;
		num1=num;
	#print num
		num=0;	
		while(s[i]!='='):
			c=""
			for j in range(0,3):
				c=c+s[i+j];
	#	print c
			if(c=="063"):
				num=num*10;
			elif(c=="010"):
				num=num*10+1;
			elif(c=="093"):
				num=num*10+2;
			elif(c=="079"):
				num=num*10+3;
			elif(c=="106"):
				num=num*10+4;
			elif(c=="103"):
				num=num*10+5;
			elif(c=="119"):
				num=num*10+6;
			elif(c=="011"):
				num=num*10+7;
			elif(c=="127"):
				num=num*10+8;
			elif(c=="111"):
				num=num*10+9;			
			i= i + 3;
		num2=num;
		res=num1+num2
		st=""
		if(res==0):
			sys.stdout.write(s+"063")
		else:
			while(res>0):
				rem=res%10
				res/=10
				if(rem==0):
					st="063"+st
				elif(rem==1):
					st="010"+st	
				elif(rem==2):
					st="093"+st	
				elif(rem==3):
					st="079"+st	
				elif(rem==4):
					st="106"+st	
				elif(rem==5):
					st="103"+st	
				elif(rem==6):
					st="119"+st	
				elif(rem==7):
					st="011"+st	
				elif(rem==8):
					st="127"+st	
				elif(rem==9):
					st="111"+st	
			sys.stdout.write(s+st+"\n")	
		try: s=sys.stdin.readline()
		except: break
pri()
sys.exit(0)

File: anarc08b.c
#include<stdio.h>
#include<string.h>
int main()
{
    while(1)
    {
    	char s[1000000];
    	scanf("%s",s);
	if(strcmp(s,"BYE")==0)
		break;
        printf("%s",s);
        int i=0;
 	    long long int num1=0,num2=0,num=0;
		while(i<strlen(s)  && isdigit(s[i]))
		{
			char c[3];
			int j;
			for(j=0;j<3;j++)
				if(i+j < strlen(s))
					c[j]=s[i+j];
				else
					break;
			c[3]='\0';
			if(strcmp(c,"063")==0)
				num=num*10;
			else if(strcmp(c,"010")==0)
				num=num*10+1;
			else if(strcmp(c,"093")==0)
				num=num*10+2;
			else if(strcmp(c,"079")==0)
				num=num*10+3;
			else if(strcmp(c,"106")==0)
				num=num*10+4;
			else if(strcmp(c,"103")==0)
				num=num*10+5;
			else if(strcmp(c,"119")==0)
				num=num*10+6;
			else if(strcmp(c,"011")==0)
				num=num*10+7;
			else if(strcmp(c,"127")==0)
				num=num*10+8;
			else if(strcmp(c,"111")==0)
				num=num*10+9;
			i+=3;
		}
		i++;
		num1=num;
		//printf("%d\n",num);
		num=0;
		while(i<strlen(s) && isdigit(s[i]))
		{
			char c[3];
			int j;
			for(j=0;j<3;j++)
				c[j]=s[i+j];
			c[j]='\0';	
	//		printf("%s\n",c);
			if(strcmp(c,"063")==0)
				num=num*10;
			else if(strcmp(c,"010")==0)
				num=num*10+1;
			else if(strcmp(c,"093")==0)
				num=num*10+2;
			else if(strcmp(c,"079")==0)
				num=num*10+3;
			else if(strcmp(c,"106")==0)
				num=num*10+4;
			else if(strcmp(c,"103")==0)
				num=num*10+5;
			else if(strcmp(c,"119")==0)
				num=num*10+6;
			else if(strcmp(c,"011")==0)
				num=num*10+7;
			else if(strcmp(c,"127")==0)
				num=num*10+8;
			else if(strcmp(c,"111")==0)
				num=num*10+9;
			i+=3;
		}
		i=0;
		num2=num;
		//printf("%d\n",num);
		num=num1+num2;
		//printf("%d\n",num);
		if(num>0){
		//int rev=(int)log10(num)+1;
        	int a[10000];
		while(num>0)
		{
			a[i]=(num%10);
			num/=10;
 		    i++;	
		}
        i--;
		while(i>=0)
		{
			int rem=a[i];
			i--;
			if(rem==0)
				printf("063");
			else if(rem==1)
				printf("010");
			else if(rem==2)
				printf("093");
			else if(rem==3)
				printf("079");
			else if(rem==4)
				printf("106");
			else if(rem==5)
				printf("103");
			else if(rem==6)
				printf("119");
			else if(rem==7)
				printf("011");
			else if(rem==8)
				printf("127");
			else if(rem==9)
				printf("111");
		}
        }
        else
        printf("063");
		printf("\n");                
    }
    return 0;
}

